src/shell/interpreter.zig

```diff
@@ -9139,15 +9262,15 @@ pub const Interpreter = struct {
 
             if (this.total_bytes_written >= SHRINK_THRESHOLD) {
                 log("IOWriter(0x{x}, fd={}) exceeded shrink threshold: truncating", .{ @intFromPtr(this), this.fd });
-                const remaining_len = this.total_bytes_written - SHRINK_THRESHOLD;
+                const remaining_len = this.buf.items.len - this.total_bytes_written;
                 if (remaining_len == 0) {
                     this.buf.clearRetainingCapacity();
                     this.total_bytes_written = 0;
                 } else {
-                    const slice = this.buf.items[SHRINK_THRESHOLD..this.total_bytes_written];
+                    const slice = this.buf.items[this.total_bytes_written..this.buf.items.len];
                     std.mem.copyForwards(u8, this.buf.items[0..remaining_len], slice);
```

Comment:
lets use `memmove()`:
```suggestion
                    bun.copy(u8, this.buf.items[0..remaining_len], slice);
```